### DISCUSSION TITLE
Configuration Help w/ Plugins

### DISCUSSION THREAD

**Discussion by aaaaaadrian** [mattmc3/antidote#180] [Q&A] [answered]
Hi there. 

First off, big fan of Antidote. I know very very very little about what's under the hood of ZSH, so things like $fpath confuse me, even after reading.

I use ZSH for scripting some simple jobs and API queries, but I like the customization of making the terminal / prompt mine.

Looking at mattmc3's Zephyr I got to thinking, is there a way to know if a plugin is duplicated or superfluous? I'd like to be as streamlined in my plugins as possible, and not have installed things I don't actually need or want.

My current .zsh_plugins.txt file is this:

```
ohmyzsh/ohmyzsh path:lib #(Framework base)

ohmyzsh/ohmyzsh path:plugins/colored-man-pages #(Enhanced man pages)
ohmyzsh/ohmyzsh path:plugins/command-not-found #(Helpful command suggestions)
ohmyzsh/ohmyzsh path:plugins/wd #(Directory shortcuts)
ohmyzsh/ohmyzsh path:plugins/alias-finder #(Alias discovery)
ohmyzsh/ohmyzsh path:plugins/colorize #(Colorized cat)
ohmyzsh/ohmyzsh path:plugins/urltools #(URL manipulation)

zsh-users/zsh-autosuggestions #(Command suggestions based on history)
zsh-users/zsh-completions #(Additional completion definitions)
marlonrichert/zsh-autocomplete #(Advanced completion system)
MichaelAquilina/zsh-you-should-use #(Command usage suggestions)
```

It looks like there are two different completions, so when I saw the Zephyr plugins, I was curious if I could replace some, or all, of my current file with these:

```
mattmc3/zephyr path:plugins/color
mattmc3/zephyr path:plugins/completion
mattmc3/zephyr path:plugins/environment
mattmc3/zephyr path:plugins/history
mattmc3/zephyr path:plugins/homebrew
mattmc3/zephyr path:plugins/macos
mattmc3/zephyr path:plugins/prompt
mattmc3/zephyr path:plugins/utility
```

Apologies for the potential basic nature and redundancy of this question. I am learning.

**Comment by mattmc3** [comment, marked as answer]
Apologies for the delay in seeing this and replying. I hope you're still a satisfied antidote user, and I'm more than happy to help here.

Zephyr is really more based on Prezto, so users familiar with Prezto should be really familiar with its plugins (aka: modules) and what they do. You seem like you prefer Oh-My-Zsh, which is perfectly fine - but mixing and matching may mean that you are overriding things, and not everything has a 1-1 correlation.

> Looking at mattmc3's Zephyr I got to thinking, is there a way to know if a plugin is duplicated or superfluous? I'd like to be as streamlined in my plugins as possible, and not have installed things I don't actually need or want.

Here's an equivalency chart, but the truth is that these aren't 1:1. You may decide you like one over the other:

| Feature                 | Oh-My-Zsh                                          | Zephyr                                 |
| ----------------------- | -------------------------------------------------- | -------------------------------------- |
| colorized man pages     | ohmyzsh/ohmyzsh path:plugins/colored-man-pages                           | mattmc3/zephyr path:plugins/color      |
| colors                  | ohmyzsh/ohmyzsh path:plugins/colorize              | mattmc3/zephyr path:plugins/color      |
| completion              | handled differently                                | mattmc3/zephyr path:plugins/completion |
| common environment vars | N/A                                                | mattmc3/zephyr path:plugins/environment |
| history                 | ohmyzsh/ohmyzsh path:lib                           | mattmc3/zephyr path:plugins/history    |
| homebrew                | ohmyzsh/ohmyzsh path:plugins/brew                  | mattmc3/zephyr path:plugins/homebrew   |
| macos                   | ohmyzsh/ohmyzsh path:plugins/macos                 | mattmc3/zephyr path:plugins/macos      |
| prompt                  | ohmyzsh/ohmyzsh path:themes/robbyrussell.zsh-theme | mattmc3/zephyr path:plugins/prompt     |
| utilities               | various                                            | mattmc3/zephyr path:plugins/utility    |

These have no equivalent in Zephyr:
- ohmyzsh/ohmyzsh path:plugins/command-not-found
- ohmyzsh/ohmyzsh path:plugins/wd
- ohmyzsh/ohmyzsh path:plugins/alias-finder
- ohmyzsh/ohmyzsh path:plugins/urltools

If you like OMZ, I would recommend sticking with it. Even though I wrote Zephyr, I don't really promote it much since it's very much a matter of personal taste. Is there a reason you are looking to switch?

Also, shameless plug - I now very much recommend using `getantidote/use-omz` as the FIRST plugin if you are using OMZ. That means your specific config becomes:

```zsh
getantidote/use-omz #(OMZ dependency resolver)
ohmyzsh/ohmyzsh path:lib #(Framework base)

ohmyzsh/ohmyzsh path:plugins/colored-man-pages #(Enhanced man pages)
ohmyzsh/ohmyzsh path:plugins/command-not-found #(Helpful command suggestions)
ohmyzsh/ohmyzsh path:plugins/wd #(Directory shortcuts)
ohmyzsh/ohmyzsh path:plugins/alias-finder #(Alias discovery)
ohmyzsh/ohmyzsh path:plugins/colorize #(Colorized cat)
ohmyzsh/ohmyzsh path:plugins/urltools #(URL manipulation)

zsh-users/zsh-autosuggestions #(Command suggestions based on history)
zsh-users/zsh-completions #(Additional completion definitions)
marlonrichert/zsh-autocomplete #(Advanced completion system)
MichaelAquilina/zsh-you-should-use #(Command usage suggestions)
```

**Comment by aaaaaadrian** [comment]
100% still satisfied with Antidote. As a beginner, the only thing that ever turns me off is if something breaks that I can't figure out how to fix.

My only thought process here was making sure I'm not using one plugin that is then canceled out by the plugin below it. IIRC this came out of some config that was causing a command to not work.

In my troubleshooting I pared my plugins down to these and it works good now

```
ohmyzsh/ohmyzsh path:lib #(Framework base)

ohmyzsh/ohmyzsh path:plugins/colored-man-pages #(Enhanced man pages)
ohmyzsh/ohmyzsh path:plugins/command-not-found #(Helpful command suggestions)
ohmyzsh/ohmyzsh path:plugins/wd #(Directory shortcuts)
ohmyzsh/ohmyzsh path:plugins/alias-finder #(Alias discovery)
ohmyzsh/ohmyzsh path:plugins/colorize #(Colorized cat)
ohmyzsh/ohmyzsh path:plugins/urltools #(URL manipulation)

zsh-users/zsh-completions #(Additional completion definitions)
MichaelAquilina/zsh-you-should-use #(Command usage suggestions)
```

But I will add the shameless plug plugin as well!

I appreciate the response! You're stellar.

**Comment by aaaaaadrian** [comment]
Actually, adding the `getantidote/use-omz` bit broke my Starship prompt and is saying it can't find my Dracula theme

```
/Users/adrian/Library/Caches/antidote/https-COLON--SLASH--SLASH-github.com-SLASH-ohmyzsh-SLASH-ohmyzsh/themes/dracula.zsh-theme:source:16: no such file or directory: /Users/adrian/Library/Caches/antidote/https-COLON--SLASH--SLASH-github.com-SLASH-ohmyzsh-SLASH-ohmyzsh/themes/lib/async.zsh
/Users/adrian/Library/Caches/antidote/https-COLON--SLASH--SLASH-github.com-SLASH-ohmyzsh-SLASH-ohmyzsh/themes/dracula.zsh-theme:19: command not found: async_init
```

```
dracula_git_async:1: command not found: async_start_worker
dracula_git_async:2: command not found: async_register_callback
dracula_git_async:3: command not found: async_job
```

**Comment by mattmc3** [comment]
@aaaaaadrian - I am very familiar with the Dracula theme. It appears that you have followed their (very poor) [manual install instructions](https://draculatheme.com/zsh) at some point and did not include the required `lib` directory. That's the error you're getting - dracula's `$ZSH/themes/lib/async.zsh` can't be found. It's got nothing to do with a use-omz bug, other than adding use-omz uncovered the bad install because you have set `ZSH_THEME=dracula` in your config.

You mention you use Starship for your theme, so I suspect perhaps you aren't even really using Dracula? If that's the case, then the simple fix is to find where you have `ZSH_THEME=dracula` in your .zshrc and simply remove that line.

If you are in fact using Dracula and want it to be installed correctly, there's two ways to fix this. You could do things the `antidote` way, and remove `ZSH_THEME=dracula` in your .zshrc and add `dracula/zsh` in your antidote .zsh_plugins.txt.

Or, you could ignore Dracula's terrible instructions and run this script instead:

```zsh
# Clone dracula theme
git clone https://github.com/dracula/zsh.git ${ZSH_CUSTOM:=$(antidote path ohmyzsh/ohmyzsh)/custom}/themes/dracula

# Properly install dracula
ln -s $ZSH_CUSTOM/themes/dracula/dracula.zsh-theme $ZSH_CUSTOM/themes
```

Hope this helps!

**Comment by aaaaaadrian** [comment]
Removing it from my .zshrc and adding it to the plugins file worked a treat. You are a swell person! I really appreciate all the help :)
